cpp/src/arrow/flight/types.h

```diff
@@ -165,8 +166,15 @@ struct ARROW_FLIGHT_EXPORT BasicAuth {
   std::string username;
   std::string password;
 
-  static Status Deserialize(const std::string& serialized, BasicAuth* out);
+  /// \brief Deserialize this message from its wire-format representation.
+  static arrow::Result<BasicAuth> Deserialize(const arrow::util::string_view& serialized);
+  /// \brief Serialize this message to its wire-format representation.
+  arrow::Result<std::string> SerializeToString() const;
 
+  ARROW_DEPRECATED("Deprecated in 8.0.0. Use Result-returning overload instead.")
+  static Status Deserialize(const arrow::util::string_view& serialized, BasicAuth* out);
```

Comment:
Hmm, not sure we should change the signature (from `string` to `string_view`) while deprecating it? Same remark for below as well.

Reply:
Good point - undid that change (I refactored the signature, then added the deprecation, should've done it the other way around.)